apps/wallet/src/ui/app/pages/swap/QuoteAssetSection.tsx

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+	Coins,
+	getUSDCurrency,
+	useBalanceConversion,
+	useRecognizedCoins,
+	useSuiBalanceInUSDC,
+} from '_app/hooks/useDeepBook';
+import { Text } from '_app/shared/text';
+import { IconButton } from '_components/IconButton';
+import { DescriptionItem } from '_pages/approval-request/transaction-request/DescriptionList';
+import { AssetData } from '_pages/swap/AssetData';
+import { MaxSlippage, MaxSlippageModal } from '_pages/swap/MaxSlippage';
+import { QuoteAssets } from '_pages/swap/QuoteAssets';
+import { useCoinMetadata } from '@mysten/core';
+import { Refresh16 } from '@mysten/icons';
+import BigNumber from 'bignumber.js';
+import clsx from 'classnames';
+import { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useSearchParams } from 'react-router-dom';
+
+import { useCoinTypeData, type FormValues } from './utils';
+
+export function QuoteAssetSection() {
+	const recognizedCoins = useRecognizedCoins();
+	const [isQuoteAssetOpen, setQuoteAssetOpen] = useState(false);
+	const [isSlippageModalOpen, setSlippageModalOpen] = useState(false);
+	const {
+		getValues,
+		formState: { isValid },
+	} = useFormContext<FormValues>();
+	const [searchParams] = useSearchParams();
```

Comment:
I'd pass this through props instead of re-fetching the search params. Otherwise this component isn't really as re-usable as it could be